src/pages/EditRequestPage.js

```diff
@@ -35,27 +35,56 @@ const defaultProps = {
     report: {},
 };
 
-function EditRequestPage(props) {
-    const parentReportAction = ReportActionsUtils.getParentReportAction(props.report);
-    const moneyRequestReportAction = ReportUtils.getMoneyRequestAction(parentReportAction);
-    const transactionDescription = moneyRequestReportAction.comment;
-    const field = lodashGet(props, ['route', 'params', 'field'], '');
-
-    function updateTransactionWithChanges(changes) {
-        // Update the transaction...
-        // eslint-disable-next-line no-console
-        console.log({changes});
+function EditRequestPage({report, route}) {
+    const parentReportAction = ReportActionsUtils.getParentReportAction(report);
+    const transactionID = lodashGet(parentReportAction, 'originalMessage.IOUTransactionID', '');
+    const transaction = TransactionUtils.getTransaction(transactionID);
+    const transactionDescription = TransactionUtils.getDescription(transaction);
+    const transactionAmount = TransactionUtils.getAmount(transaction);
+    const transactionCurrency = TransactionUtils.getCurrency(transaction);
+    const transactionCreated = TransactionUtils.getCreated(transaction);
+    const fieldToEdit = lodashGet(route, ['params', 'field'], '');
 
-        // Note: The "modal" we are dismissing is the MoneyRequestAmountPage
+    // Update the transaction object and close the modal
+    function editTransaction(transactionChanges) {
+        IOU.editMoneyRequest(transactionID, report.reportID, transactionChanges);
         Navigation.dismissModal();
     }
 
-    if (field === CONST.EDIT_REQUEST_FIELD.DESCRIPTION) {
+    if (fieldToEdit === CONST.EDIT_REQUEST_FIELD.DESCRIPTION) {
         return (
             <EditRequestDescriptionPage
                 defaultDescription={transactionDescription}
-                onSubmit={(changes) => {
-                    updateTransactionWithChanges(changes);
+                onSubmit={(transactionChanges) => {
+                    editTransaction(transactionChanges);
+                }}
+            />
+        );
+    }
+
+    if (fieldToEdit === CONST.EDIT_REQUEST_FIELD.DATE) {
+        return (
+            <EditRequestCreatedPage
+                defaultCreated={transactionCreated}
+                onSubmit={(transactionChanges) => {
+                    editTransaction(transactionChanges);
+                }}
+            />
+        );
+    }
+
+    if (fieldToEdit === CONST.EDIT_REQUEST_FIELD.AMOUNT) {
+        return (
+            <EditRequestAmountPage
+                defaultAmount={transactionAmount}
+                defaultCurrency={transactionCurrency}
+                reportID={report.reportID}
+                onSubmit={(transactionChanges) => {
```

Comment:
The amount page returns the amount as full number without any key now so `transactionChanges` holds amount as `9` when its `$9.00` for example